@here/harp-mapview/lib/image/ImageCache.ts

```diff
@@ -271,46 +281,56 @@ export class ImageCache {
             return imageItem.loadingPromise;
         }
 
-        const imageLoader = new THREE.ImageLoader();
-
-        imageItem.loadingPromise = new Promise(resolve => {
-            logger.debug(`Loading image: ${imageItem.url}`);
-            if (imageItem.cancelled === true) {
-                logger.debug(`Cancelled loading image: ${imageItem.url}`);
-                resolve(undefined);
-            } else {
-                imageLoader.load(
-                    imageItem.url,
-                    image => {
-                        logger.debug(`... finished loading image: ${imageItem.url}`);
-                        if (imageItem.cancelled === true) {
-                            logger.debug(`Cancelled loading image: ${imageItem.url}`);
+        if (imageItem.htmlElement) {
+            imageItem.loadingPromise = new Promise(resolve => {
+                if (
+                    imageItem.htmlElement instanceof HTMLImageElement &&
+                    !imageItem.htmlElement.complete
+                ) {
+                    imageItem.htmlElement.addEventListener(
+                        "load",
+                        (this.createOnImageLoaded(imageItem, resolve) as unknown) as (
+                            this: HTMLImageElement,
+                            ev: Event
+                        ) => any
+                    );
+                    imageItem.htmlElement.addEventListener("error", err => resolve(undefined));
+                } else {
+                    this.createOnImageLoaded(
+                        imageItem,
+                        resolve
+                        // cast to be a HtmlCanvasElement, as undefined and HtmlImageElement are
+                        // already excluded from the conditions above
+                    )(imageItem.htmlElement as HTMLCanvasElement);
+                }
+            });
+        } else {
+            const imageLoader = new THREE.ImageLoader();
+
+            imageItem.loadingPromise = new Promise(resolve => {
+                logger.debug(`Loading image: ${imageItem.url}`);
+                if (imageItem.cancelled === true) {
+                    logger.debug(`Cancelled loading image: ${imageItem.url}`);
+                    resolve(undefined);
+                } else {
+                    imageLoader.load(
+                        imageItem.url,
+                        // Loading events no longer supported
```

Comment:
This comment is confusing. I found nothing in the three.js docs that it is no longer supported

Reply:
Seems to be some remainer of the former code, just copied it somewhere else,... I agree, removing it